### ISSUE TITLE
Add-On to Suggestion #585 that we need for subscriptions and usergroups

### ISSUE THREAD

**Issue by spirogg** [PHPfox-Official/phpfox-v4-feature-requests#880]
@harrison05 based on #585 

- Can we add the ability to set the limits to all Apps - NOT JUST groups, pages, events 

**PLEASE ADD Limits to all these apps **

Music: number of music uploads
Photos: number of photo uploads
Videos: number of video uploads
Quizzes: number of quizzes a user can create
Polls: number of polls a user can create 
Blogs: number of blogs a user can create 
Pages: number of pages a user can create 
Groups: number of groups a user can create 
Events: number of events a user can create 
Marketplace: number of items a user can add for sale

Note:: when a user creates Groups and Pages the limits should also be used for photos music blogs etc in groups and pages section.

- Since we already have a feature in AdminCP : how many MB of space a usergroup is allowed to use? this should also be included in my idea. 

- If a site admin wishes to use subscription plans, we then turn on the feature to set the limits in each app for that user group - plan. 

- But if an admin does not use the subscriptions feature, we would still be able to add limits in each app  for user groups..

- Depending on the website's owner(s) they  may want to create a Private Only Community ( Friends Only) or Work Employee's Only or Family Only 
- So we would still need to have the same option to limit each app in there respective settings in admincp 

QUESTION - Have you thought about how the usergroup will get notification when limit is reached ?? 

- My Idea a Popup when that apps limit has been reached and maybe even an email sent to that user when they are running low on storage  ?

_Hello {username}, you have reach your maximum ie. ( storage space, Pages, Groups, blogs, events, photos, music uploads, video uploads, storage space etc. )
 which ever limit they reach that individual app sends the Popup Notification on Screen with a link to upgrade to the next level up ( paid subsciption )_ 

- If we don't use subscriptions, user still gets notification via popup 

Hello {username}, Your limit for the app has been reached please **_click here_** to contact support. 

** We would need to add in admincp a section for this Limit setting.

(Do you want to use limits with Subsciptions Yes or NO)
If No 
- when a user clicks the link in the popup that says click here to contact support, then they are presented with a Popup form they can fill out to request support.
  In Admincp the admin will then get a notification (where we get notifications already) to review the request and to upgrade this user to a higher limit or deny them . When admin clicks on notification they are able to read and reply to this request. 

Users will then receive notification in front end of script (where users see all notifications already)  with approval and they will click on it and it will take them to the account page to show them there new limits or the admin can deny the request and the user will get a page with the reason for denial..

 
in Admincp the Limit app option should be :

- if we set this option to =Yes (Popup will be to upgrade to higher subscription)

- if we set this option to =No  (popup will be to click here to contact support)

this will be good for all admins no matter what we use our PhpFox script community for..

**Note: This should be a standard feature NOT an extra app from a third party.   

Thanks and hope you can Reconsider your thoughts on how to finish #585 
Would be really nice with the features I have outlined above for you

** Please give me your thoughts @harrison05 (PhpFox Team) 

Anyone else 
And @matthewpomar who came up with the initial thought for this feature request. 
Thank you @matthewpomar

PS.

- I think this would be the best solution

- each app has its own limits we can set per user group

- when a limit has reach the end a popup for that app shows a link to either upgrade subscription
 or
To Email support via a Popup form (for admins not using PhpFox with subscriptions) but they still need to limit each app for there business - friends only community needs. 

- Thanks so much, 
Spirog 

I hope you can make this feature a priority because we really need this feature soon.

**Comment by spirogg** [comment]
@harrison05
To take this feature even further to be an awesome and wow idea 😁

We can create features - limits page in users profile/ section
- Showing the features and limits to Their account 

** Kind of like the activity point page. 

***The Page should look like this below: 

Your plan: Registered {Usergroup}

(Available Storage and Limits)
IDEA BELOW is just simple you can make it nice looking with icons for each app :)

- Allowed Storage 100mb   used (50%) or  50mb 

- Allowed Photos 10                used 2

- Allowed Videos 10                used 2

- Allowed Blogs 1                    used 1

- Allowed Events 5                  used 2

- Allowed Pages 2                   used 2

- Allowed Groups 2                used 2

- Allowed Quizzes 5               used 2

- Allowed Polls 5                    used 2

- Allowed Market listings 5    used 3

Please Note: ( If we don't allow Marketplace, Blogs or Video uploads or any other app, Then this page should hide ( not show ) the Marketplace, Blog and videos app limits on this (Available Storage and Limits) Page.



PLEASE: 
- The Example above would make a nice add-on feature to the limits feature and really wow PhpFox customers - Their users.  

Thanks again, 
I sure hope you are able to make this soon. 

****This feature should make PhpFox stand out even more than any other community script..****

- This is an AWESOME IDEA now that I think of it more and more. 

 - The users will know exactly what they have and will not have no surprises or they will Not get mad...
 why ?? I can't upload no more??? 🤔
Or 
why I can't add blogs or more photos  no more ?? 😢

- This will show exactly what they are allowed and they can see how much of Their limits they already used. 😁

- Plus this can bring more Money - Sales for Just this feature to PhpFox Script more 💲for everyone 🎁🎉🌈

Regards,
Spirog

**Comment by harrison05** [comment]
Your suggestions sound good. But it must be reviewed by our team, only me can't decide to accept or not. I will inform you if have any updates.

Thank you.

**Comment by PhpFoxJohnJr** [comment]
Didn't see this request as I just did one for limits on groups and pages by User Group Membership.

**Comment by spirogg** [comment]
@PhpFoxJohnJr Yes wish they would consider this soon. Hope your doing well my friend. 

@harrison05 Hello long time my friend. I wish everyone is well. Hope your team can consider this feature again. 

We need more control of limits please. 

Best Regards,

Spiro
